test/jdk/java/util/logging/LoggingDeadlock5.java

```diff
@@ -115,7 +118,13 @@ static void assertNoDeadlock(Handler handler) throws InterruptedException, Unsup
     }
 
     private static class DeadLocker {
-        private final static Duration JOIN_WAIT = Duration.ofMillis(500);
+        // Since this is used to self-test for an expected deadlock, it will
+        // delay the test by at least this duration (so being overly large is
+        // a potential issue). The deadlock is set up so it should occur almost
+        // immediately (if it occurs), but tests are run under very high loads
+        // in higher tiers, so it's necessary to be a bit pessimistic here.
+        private final static Duration JOIN_WAIT =
+                Duration.ofMillis(Utils.adjustTimeout(2000));
```

Comment:
Just checking, adjustTimeout does scaling based on the timeout factor given to jtreg?

What happens to the expected test running time to the test in lower tiers without as high loads?

Reply:
Note, the timeout factor also adjusts the jtreg timeout for the entire test. The adjustTimeout() method allows internal test timeouts to scale along with the jtreg timeout.

Reply:
> Hi Joe, yes `adjustTimeout` will scale based on the jtreg timeout factor. I believe the behaviour is to multiply whatever hardcoded timeout is passed by the timeout factor. On lower tiers, in our CI, I beleive it means the test will have to wait for 8s before it can assert that a deadlock is detected. That should be way off the default jtreg timeout - which IIRC is 480s on lower tiers.

Catching up on PR comments, to make my concerns more explicit: it is best is we don't pessimize the common case of increasing the _minimum_ run time of the test in order to make the test more robust in highly-contended situations.

Reply:
Right. But it's still way better than hard to diagnose intermittent failures. If this [the 2s x timeout factor minimum time] becomes an issue - I guess we could update the test to only test the case where there should no longer be a deadlock - and remove the self-test that proves that a deadlock would be detected if there was one.
That would need to be carried out in a different PR.